apps/site/assets/ts/schedule/__tests__/LineDiagramTest.tsx

```diff
@@ -129,7 +130,64 @@ describe("LineDiagram", () => {
   );
 
   it("uses the route color", () => {
-    const line = wrapper.find(".m-schedule-line-diagram__line").first();
+    const line = wrapper.find(".m-schedule-diagram__lines").first();
     expect(line.prop("style")!.color).toBe("#F00B42");
   });
 });
+
+describe("LineDiagram with branches", () => {
+  let wrapper: ReactWrapper;
+  beforeEach(() => {
+    wrapper = mount(
+      <LineDiagram
+        lineDiagram={lineDiagramWithBranchesData as LineDiagramStop[]}
+        route={route}
+        directionId={directionId}
+        routePatternsByDirection={
+          routePatternsByDirection as RoutePatternsByDirection
+        }
+        services={[]}
+        ratingEndDate="2020-03-14"
+        stops={{ 0: stops, 1: stops }}
+      />
+    );
+  });
+
+  afterEach(() => {
+    wrapper.unmount();
+  });
+
+  it("renders and matches snapshot", () => {
+    expect(wrapper.debug()).toMatchSnapshot();
+  });
+
+  it("shows branch name", () => {
+    expect(wrapper.find(".u-small-caps").text()).toEqual("Lowell Two Line");
+  });
+
+  it("renders different parts of branches", () => {
+    const terminus = wrapper
+      .find(".m-schedule-diagram__line--terminus")
+      .first()
+      .html();
+    const merge = wrapper
+      .find(".m-schedule-diagram__line--merge")
+      .first()
+      .html();
+    const stop = wrapper
+      .find(".m-schedule-diagram__line--stop")
+      .first()
+      .html();
+    const line = wrapper
+      .find(".m-schedule-diagram__line--line")
+      .first()
+      .html();
+
+    expect(terminus).not.toEqual(stop);
+    expect(terminus).not.toEqual(line);
+    expect(terminus).not.toEqual(merge);
+    expect(stop).not.toEqual(merge);
+    expect(stop).not.toEqual(line);
+    expect(line).not.toEqual(merge);
```

Comment:
I'm not quite sure what behavior this test aims to exercise — could we maybe use some specifics from the fixture data? For example: station X should appear as a terminus, station Y a merge, station Z a normal stop...